File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.swift
import Foundation

func solution(_ price:Int, _ money:Int, _ count:Int) -> Int64{
    var answer:Int64 = 0
    
    for i in 1...count {
        answer += Int64(price * i)
    }
    answer -= Int64(money)
    
    return answer > 0 ? answer : 0
}

File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.java
import java.util.*;
class Solution {
    public long solution(int price, int money, int count) {
        long answer = 0;
        long totalprice = 0;
        for(int i=1; i<=count; i++) {
            totalprice += price * i;
        }
        
        if (totalprice - money <= 0) answer = 0;
        else  answer = Math.abs(totalprice - money);
        return answer;
    }
}